packages/common/src/utils/convertCustomDimensionsToYaml.ts

```diff
@@ -1,12 +1,63 @@
-import type { DbtColumnLightdashDimension } from '../types/dbt';
-import { type CustomSqlDimension, friendlyName } from '../types/field';
+import type { DbtColumnLightdashAdditionalDimension } from '../types/dbt';
+import { ParameterError } from '../types/errors';
+import {
+    BinType,
+    type CustomBinDimension,
+    type CustomSqlDimension,
+    DimensionType,
+    friendlyName,
+} from '../types/field';
+import { type WarehouseClient } from '../types/warehouse';
+import {
+    getCustomRangeSelectSql,
+    getFixedWidthBinSelectSql,
+} from './customDimensions';
 
 export const convertCustomDimensionToDbt = (
     field: CustomSqlDimension,
-): DbtColumnLightdashDimension => ({
+): DbtColumnLightdashAdditionalDimension => ({
     label: friendlyName(field.name),
-    name: field.id,
     description: '', // TODO :: leave it here so that customer can easily fill in?
     type: field.dimensionType,
     sql: field.sql,
 });
+
+export const convertCustomBinDimensionToYaml = ({
+    customDimension,
+    baseDimensionSql,
+    warehouseClient,
+}: {
+    customDimension: CustomBinDimension;
+    baseDimensionSql: string;
+    warehouseClient: WarehouseClient;
+}): DbtColumnLightdashAdditionalDimension => {
+    switch (customDimension.binType) {
+        case BinType.CUSTOM_RANGE:
+            return {
+                label: friendlyName(customDimension.name),
+                description: '',
+                type: DimensionType.STRING,
+                sql: getCustomRangeSelectSql({
+                    binRanges: customDimension.customRange || [],
+                    baseDimensionSql,
+                    warehouseClient,
+                }),
+            };
+        case BinType.FIXED_WIDTH:
+            return {
+                label: friendlyName(customDimension.name),
+                description: '',
+                type: DimensionType.STRING,
+                sql: getFixedWidthBinSelectSql({
+                    binWidth: customDimension.binWidth || 1,
+                    baseDimensionSql,
+                    warehouseClient,
+                }),
+            };
+        case BinType.FIXED_NUMBER:
+            throw new ParameterError('Fixed number bin type not supported');
```

Comment:
can we do `NotImplementedError` here?

https://github.com/lightdash/lightdash/blob/e4c4a88f69b852fd962d19ac4783401400d94913/packages/frontend/src/components/Explorer/WriteBackModal/index.tsx#L56

https://github.com/lightdash/lightdash/blob/832f527e8f50f7946deb60341ce2331c04008ca0/packages/common/src/types/filterGrammarConversion.ts#L81